packages/components/src/internal/components/domainproperties/ScannableOption.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, memo, useCallback } from 'react';
+import { Col, FormControl, Row } from 'react-bootstrap';
+import { DomainFieldLabel } from './DomainFieldLabel';
+import { createFormInputId, createFormInputName } from './actions';
+import { DOMAIN_FIELD_SCANNABLE_OPTION } from './constants';
+import { isFieldFullyLocked } from './propertiesUtil';
+import { ITypeDependentProps } from './models';
+
+export interface ScannableProps extends ITypeDependentProps{
+    scannable?: boolean;
+    appPropertiesOnly?: boolean;
+    showScannableOption?: boolean;
```

Comment:
should these two (or all three of them) be required props?

Reply:
then we would need to update text and number fields in domains in other places, I think defaulting them to false if not present is suffcient. Especially since we are only supporting the strict case of SampleTypes in our Apps at this point.

Reply:
sounds good